cinema/views.py

```diff
@@ -101,8 +94,24 @@ def get_serializer_class(self):
         if self.action == "retrieve":
             return MovieDetailSerializer
 
+        if self.action == "upload_image":
+            return MovieImageSerializer
+
         return MovieSerializer
 
+    @action(
+        detail=True,
+        methods=["POST"],
+        url_path="upload-image",
+    )
+    def upload_image(self, request, pk=None):
+        movie = self.get_object()
+        serializer = self.get_serializer(movie, data=request.data)
+        if serializer.is_valid(raise_exception=True):
```

Comment:
If you use` raise_exception=True`, the exception are raised under the hood. You don't need to do it here. Returning Response with errors is redundant here.